tests/testthat/test-carehomes-multistrain.R

```diff
@@ -764,3 +760,248 @@ test_that("calculate Rt with both second variant and vaccination", {
+                                    strain_rel_gamma_A = c(1, 5),
+                                    strain_transmission = c(1, 2, 3)),
+               "1 or 3")
+  expect_error(carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                                    strain_transmission = c(1, 1),
+                                    strain_rel_gamma_A = c(2, 5)),
+               "must be 1")
+  expect_error(carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                                    strain_transmission = c(1, 1),
+                                    strain_rel_gamma_A = c(1, -1)),
+               "non-negative")
+})
+
+
+test_that("carehomes_parameters_progression works as expected", {
+  gammas <- c("gamma_A", "gamma_P", "gamma_C_1", "gamma_C_2")
+  defaults <- c(1 / 2.88, 1 / 1.68, 1 / 2.14, 1 / 1.86)
+  expect_equal(
+    as.numeric(carehomes_parameters_progression(1, 1, 1, 1)[gammas]),
+    defaults
+  )
+  expect_equal(
+    as.numeric(carehomes_parameters_progression(2, 2, 2, 2)[gammas]),
+    defaults * 2
+  )
+  expect_equal(
+    matrix(unlist(carehomes_parameters_progression(1:4, 1:4, 1:4, 1:4)[gammas]),
+           ncol = 4),
+    vapply(defaults, function(x) x * 1:4, numeric(4))
+  )
+})
+
+
+test_that("Relative gamma = 1 makes no difference", {
+  p1 <- carehomes_parameters(sircovid_date("2020-02-07"), "england")
+  p2 <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                             strain_transmission = c(1, 1))
+  p3 <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                             strain_transmission = c(1, 1))
+  np <- 10
+  mod1 <- carehomes$new(p1, 0, np, seed = 1L)
+  mod2 <- carehomes$new(p2, 0, np, seed = 1L)
+  mod3 <- carehomes$new(p3, 0, np, seed = 1L)
+  end <- sircovid_date("2020-03-31") / p1$dt
+
+  mod1$set_index(carehomes_index(mod1$info())$run)
+  mod2$set_index(carehomes_index(mod2$info())$run)
+  mod3$set_index(carehomes_index(mod3$info())$run)
+
+  initial1 <- carehomes_initial(mod1$info(), 1, p1)
+  initial2 <- carehomes_initial(mod1$info(), 1, p2)
+  initial3 <- carehomes_initial(mod1$info(), 1, p3)
+
+  res1 <- mod1$run(end)
+  res2 <- mod2$run(end)
+  res3 <- mod3$run(end)
+
+  expect_equal(res2, res1)
+  expect_equal(res3, res1)
+})
+
+
+test_that("Higher rate variant has lower Rt", {
+  ## rate is 1.5 times ref
+  p <- carehomes_parameters(sircovid_date("2020-02-07"), "england",
+                            strain_transmission = c(1, 1),
+                            strain_rel_gamma_A = c(1, 1.5),
+                            strain_rel_gamma_P = c(1, 1.5),
+                            strain_rel_gamma_C_1 = c(1, 1.5),
+                            strain_rel_gamma_C_2 = c(1, 1.5),
+                            strain_seed_date =
+                              rep(sircovid_date("2020-02-07"), 2),
+                            strain_seed_rate = c(10, 0))
+  p$gamma_A[1] <- p$gamma_A[1] * 1.5
```

Comment:
I don't think you need to do this - even if gamma_A is unchanged for strain 1 but is 1.5 times higher for strain 2 then the overall reproduction number should be lower